docs/src/main/tut/chisel3/interfaces-and-connections.md

```diff
@@ -129,3 +137,18 @@ class ConsumingData extends Module {
   // do something with io.readyValid.bits
 }
 ```
+
+`DecoupledIO` is a ready-valid interface with the *convention*
+that there are no guarantees placed on de-asserting `ready` or `valid` or
+on the stability of `bits` after a clock cycle without data transfer.
+That means `ready` and `valid` can also be de-asserted without a data transfer.
+Specifically, `ready` and `valid` may be deasserted and `bits` may change
+after a clock cycle without a data transfer.
+
+`IrrevocableIO` is a ready-valid interface with the *convention*
+that the value of `bits` will not change while `valid` is asserted and `ready` is deasserted.
+Also the consumer shall keep `ready` asserted after a cycle where `read`
+was high and `valid` was low.
+Note that the *irrevocable* constraint *is only a convention* and cannot
+be enforced by the interface.
```

Comment:
> Note that the *irrevocable* constraint *is only a convention* and cannot be enforced by the interface.

This statement is true of most interfaces (including AXI and TileLink). What I would note is that we don't generate checkers/assertions.